src/System.Private.CoreLib/shared/System/Threading/Timer.cs

```diff
@@ -290,6 +290,10 @@ private void FireNextTimers()
 
         #region Queue implementation
 
+        private long _count;
+
+        public long Count => Volatile.Read(ref _count);
```

Comment:
On 32-bit, is a Volatile.Read sufficient here to avoid a torn-write from the non-volatile/interlocked ++ and -- updated _count?

Reply:
Nope, will fix

Reply:
The read is ok; as the Volatile api does a `Interlocked.CompareExchange` on 32bit

https://github.com/dotnet/coreclr/blob/730c3c5c66183418adbbc63acd40d261c9ce23ed/src/System.Private.CoreLib/shared/System/Threading/Volatile.cs#L92-L98

Not sure the writes are ok though?

Reply:
> Not sure the writes are ok though?

Right, that's why I said "torn-write" :)